fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ParameterVisitorBatchDAO.java

```diff
@@ -135,73 +129,45 @@ public ParameterVisitorBatchDAO(Connection c, String adminSchemaName, String tab
-                + "comp1_str, comp1_number, comp1_date, comp1_token, comp1_quantity, comp1_latlng, "
-                + "comp2_str, comp2_number, comp2_date, comp2_token, comp2_quantity, comp2_latlng, "
-                + "comp3_str, comp3_number, comp3_date, comp3_token, comp3_quantity, comp3_latlng"
-                + ") VALUES (" + adminSchemaName + ".sv_tenant_id,?,?,  ?,?,?,?,?,?,  ?,?,?,?,?,?,  ?,?,?,?,?,?)"
-                :
-                "INSERT INTO " + tablePrefix + "_composites (parameter_name_id, logical_resource_id, "
-                + "comp1_str, comp1_number, comp1_date, comp1_token, comp1_quantity, comp1_latlng, "
-                + "comp2_str, comp2_number, comp2_date, comp2_token, comp2_quantity, comp2_latlng, "
-                + "comp3_str, comp3_number, comp3_date, comp3_token, comp3_quantity, comp3_latlng"
-                + ") VALUES (?,?,  ?,?,?,?,?,?,  ?,?,?,?,?,?,  ?,?,?,?,?,?)";
-        composites = c.prepareStatement(insertComposite);
-
-        // Resource level string attributes
-        String insertResourceString = multitenant ?
-                "INSERT INTO resource_str_values (mt_id, parameter_name_id, str_value, str_value_lcase, logical_resource_id) VALUES (" + adminSchemaName + ".sv_tenant_id,?,?,?,?)"
-                :
-                "INSERT INTO resource_str_values (parameter_name_id, str_value, str_value_lcase, logical_resource_id) VALUES (?,?,?,?)";
-        resourceStrings = c.prepareStatement(insertResourceString);
+        insertLocation = multitenant ? "INSERT INTO " + tablePrefix + "_latlng_values (mt_id, parameter_name_id, latitude_value, longitude_value, logical_resource_id, composite_id) VALUES (" + adminSchemaName + ".sv_tenant_id,?,?,?,?,?)"
+                : "INSERT INTO " + tablePrefix + "_latlng_values (parameter_name_id, latitude_value, longitude_value, logical_resource_id, composite_id) VALUES (?,?,?,?,?)";
 
-        // Resource level date attributes
-        String insertResourceDate = multitenant ?
-                "INSERT INTO resource_date_values (mt_id, parameter_name_id, date_start, date_end, logical_resource_id) VALUES (" + adminSchemaName + ".sv_tenant_id,?,?,?,?)"
+        // System level string attributes
+        String insertSystemString = multitenant ?
+                "INSERT INTO str_values (mt_id, parameter_name_id, str_value, str_value_lcase, logical_resource_id) VALUES (" + adminSchemaName + ".sv_tenant_id,?,?,?,?)"
                 :
-                "INSERT INTO resource_date_values (parameter_name_id, date_start, date_end, logical_resource_id) VALUES (?,?,?,?)";
-        resourceDates = c.prepareStatement(insertResourceDate);
+                "INSERT INTO str_values (parameter_name_id, str_value, str_value_lcase, logical_resource_id) VALUES (?,?,?,?)";
+        systemStrings = c.prepareStatement(insertSystemString);
 
-        // Resource level token attributes
```

Comment:
Just double-checking...never any need to store System-level token attributes now?  I think I knew that the token values would be in the common-token-values table now, but I was thinking we still needed a system-level tokens table which links those common values to the resources which have them.

Reply:
oh thats right...we write them all at the end of the transaction now (and I assume we do the same with system ones now)...remind me where we register that "transaction hook" again (or what the right name is for this feature)?

Reply:
found it.  lines 380-384.  I'm good now.